### PR TITLE
When creating an order, and assigning a customer, fill all the customer inputs in the shipping/billing address form

### LINKED ISSUE #9140: Address field not updating correctly for existing customer when creating new order in backend
## Description

When creating a new order in the backend and choosing an existing customer from a shop/hub's customer list for the 'customer details' section, the 'province' (in Canada - may be state/county in other instances) field is not being updated when the address info gets automatically filled. 

## Expected Behavior

The province field should be updated to the province that is associated with the customers address on file.  

## Actual Behaviour

The province field remained the default selection (Alberta in Canada because the options sort alphabetically).

## Steps to Reproduce

1. Create a new order in the backend.
2. Add products, and then go to customer details to fill in the customer info.
3. Select a customer from the existing customer list that has a stored address with a province other than the default selection (i.e. other than Alberta in Canada)
4. See that the province field did not update correctly when the other address info was automatically filled.

## Animated Gif/Screenshot

Users can manually updated the Province field if they notice it is incorrect.  However if they are not aware of this issue they may not notice, and assume this field is updated when the other address info is automatically filled.

## Severity

bug-s4?  If the error is not noticed by the user and updated manually it can lead to incorrect taxes being applied.

## Your Environment

* Version used: OFN Canada
* Browser name and version: Firefox 99.0.1
* Operating System and version (desktop or mobile): MacOS Monterey Version 12.3.1 desktop

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?
As we've introduced a new select component (named `tom-select`, see PR #8814), address fields city and state were not filled correctly after selecting a customer into an order (see original issue: #9140).
This was due to the fact that we used to use a AngularJs directive (`app/assets/javascripts/admin/orders/directives/customer_search_override.js.coffee`), that fill the city and state with:
```js
$(attribute_wrapper + 'state_id').select2 'val', data['state_id']
```

➡️ `select2` was the old select component, and cannot be handled by `tom-select`

This PR then: 
 - Remove the `customerSearchOverride` directive
 - and remplace with a StimulusJs controller (`app/webpacker/controllers/select_customer_controller.js`)
 - This stimulusjs controller can now use the `tom_select_controller` to set the value on the city/state fields.

Closes #9140

#### What should we test?
Well described in the linked issue

1. Create a new order in the backend.
2. Add products, and then go to customer details to fill in the customer info.
3. Select a customer from the existing customer list that has a stored address with a province other than the default selection (i.e. other than Alberta in Canada)
4. See that the city and state fields update correctly when the other address info was automatically filled, as well as the other fields for both shipping and billing address (and doest not stay on the first item of the dropdown items list). 

Capture d’écran 2022-05-05 à 09 38 18

#### Release notes

Changelog Category: User facing changes 

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/webpacker/controllers/tom_select_controller.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
-import { Controller } from "stimulus"
-import TomSelect from "tom-select"
+import { Controller } from "stimulus";
+import TomSelect from "tom-select";
```
Comment (mkllnk) [inline app/webpacker/controllers/tom_select_controller.js:2]: Thank you for formatting in a separate commit. :pray:

File: spec/system/admin/order_spec.rb [hunk 2]
```diff
@@ -392,14 +392,18 @@ def new_order_with_distribution(distributor, order_cycle)
     expect(page).to have_selector '#select-customer'
 
     # And I select that customer's email address and save the order
-    select2_select customer.email, from: 'customer_search_override', search: true
+    tomselect_search_and_select customer.email, from: 'customer_search_override'
```
Comment (mkllnk) [inline spec/system/admin/order_spec.rb:395]: I thought we could just use `select` now.

https://github.com/openfoodfoundation/openfoodnetwork/pull/8814/files#diff-5cad13c68b9482412216548c5f4976e9db9f062376532324dbe4f7790de03394

Does it not trigger tom actions? Or should we take a step back and subscribe to `onChange`?

File: spec/support/request/web_helper.rb [hunk 3]
```diff
@@ -127,6 +127,12 @@ def close_select2
     page.find(:css, 'body').click
   end
 
+  def tomselect_search_and_select(value, options)
+    page.find("[name='#{options[:from]}']").sibling(".ts-wrapper").click
+    page.find(:css, '.ts-dropdown input.dropdown-input').set(value)
+    page.find(:css, '.ts-dropdown .ts-dropdown-content .option', text: value).click
```
Comment (mkllnk) [inline spec/support/request/web_helper.rb:133]: So what if a user doesn't click but uses the keyboard or some other device to choose a value?

**Review by mkllnk — APPROVED** [review]
Nice! :sunglasses: 

It's a shame that the new code isn't shorter than the old one. I was hoping that replacing AngularJS would make things simpler, too.

Anyway, this seems to solve a really annoying issue. So I'll approve this while also wondering if we could improve it.

File: spec/system/admin/order_spec.rb (hunk 2 above)
Comment (jibees) [reply spec/system/admin/order_spec.rb:395]: > I thought we could just use `select` now.

Yes, but this one is a particular case of selection: data are retrieved via request from user's input:
Capture d’écran 2022-05-10 à 10 21 30

That's why, in `tomselect_search_and_select` method, we emulate user input and click.

File: spec/support/request/web_helper.rb (hunk 3 above)
Comment (jibees) [reply spec/support/request/web_helper.rb:133]: `tomselect_search_and_select` is not here to test `tom-select` but to create a helper method that search and select from a tom-select component via Ajax request:

**Review by mkllnk — APPROVED** [review]
Thank you for your patience and super clear replies! :hugs:

**Comment by RachL** [comment]
This is looking good! 

Merging :) 

image

**Comment by jibees** [comment]
Reverted because of an error when customer has no bill address.

**Pull request merged**